tools/cli/NearObjectCli.cxx

```diff
@@ -276,70 +280,80 @@ ValidateNonEnumParameterValues(NearObjectCliData& cliData)
     if (parametersData.rangeDataNotificationProximityFar.has_value()) {
         if (parametersData.rangeDataNotificationConfig.has_value() && parametersData.rangeDataNotificationConfig.value() == RangeDataNotificationConfiguration::EnableInProximityRange) {
             if (parametersData.rangeDataNotificationProximityNear.has_value() && parametersData.rangeDataNotificationProximityFar.value() < parametersData.rangeDataNotificationProximityNear.value()) {
-                std::cerr << "Invalid RangeDataNotificationProximityFar" << std::endl;
+                std::cerr << "Invalid RangeDataNotificationProximityFar. Should be greater than or equal to RangeDataNotificationProximityNear (default: " << +DefaultRangeDataNotificationProximityNear << ")." << std::endl;
             }
         }
     }
 
     // PreambleCodeIndex
     if (parametersData.preambleCodeIndex.has_value()) {
         if (!parametersData.prfMode.has_value() || (parametersData.prfMode.has_value() && parametersData.prfMode.value() == PrfModeDetailed::Bprf62MHz)) { // Either BPRF is set or PRF_MODE is left at default (BPRF)
-            if (parametersData.preambleCodeIndex.value() < 9 || parametersData.preambleCodeIndex.value() > 12) {
-                std::cerr << "Invalid PreambleCodeIndex. Expected value range of 9-12 in BPRF mode" << std::endl;
+            if (parametersData.preambleCodeIndex.value() < MinimumPreambleCodeIndexBprf || parametersData.preambleCodeIndex.value() > MaximumPreambleCodeIndexBprf) {
+                std::cerr << "Invalid PreambleCodeIndex. Expected value range of " << +MinimumPreambleCodeIndexBprf << "-" << +MaximumPreambleCodeIndexBprf << " in BPRF mode." << std::endl;
             }
         } else { // HPRF mode
-            if (parametersData.preambleCodeIndex.value() < 25 || parametersData.preambleCodeIndex.value() > 32) {
-                std::cerr << "Invalid PreambleCodeIndex. Expected value range of 25-32 in HPRF mode" << std::endl;
+            if (parametersData.preambleCodeIndex.value() < MinimumPreambleCodeIndexHprf || parametersData.preambleCodeIndex.value() > MaximumPreambleCodeIndexHprf) {
+                std::cerr << "Invalid PreambleCodeIndex. Expected value range of " << +MinimumPreambleCodeIndexHprf << "-" << +MaximumPreambleCodeIndexHprf << " in BPRF mode." << std::endl;
             }
         }
     }
 
     // SfdId
     if (parametersData.sfdId.has_value()) {
         if (!parametersData.prfMode.has_value() || (parametersData.prfMode.has_value() && parametersData.prfMode.value() == PrfModeDetailed::Bprf62MHz)) { // Either BPRF is set or PRF_MODE is left at default (BPRF)
-            if (parametersData.sfdId.value() != 0 && parametersData.sfdId.value() != 2) {
-                std::cerr << "Invalid SfdId. Expected values of 0 or 2 in BPRF mode" << std::endl;
+            if (std::ranges::find(SfdIdValuesBprf, parametersData.sfdId.value()) == SfdIdValuesBprf.end()) {
```

Comment:
nit: should be consistent with iterator access

```suggestion
            if (std::ranges::find(SfdIdValuesBprf, parametersData.sfdId.value()) == std::ranges::end(SfdIdValuesBprf)) {
```

Will be nice to get the C++23 [`std::ranges:contains`](https://en.cppreference.com/w/cpp/algorithm/ranges/contains) which basically implements exactly this.